src/input/sources/http.rs

```diff
@@ -82,6 +82,12 @@ impl HttpRequest {
             .await
             .map_err(|e| AudioStreamError::Fail(Box::new(e)))?;
 
+        if !resp.status().is_success() {
+            return Err(AudioStreamError::HttpRequestFailed(
+                resp.status().to_string(),
```

Comment:
As above, please convert into an `AudioStreamError::Fail`. You can either place a string in here directly (as it should be coerceable to `Box<dyn Error>`) or you can perhaps setup a local `Error` struct which stores the status code and/or body for use in a `Display` impl.